File: php/5-longest-palindromic-substring.php
<?php
class Solution {
    /**
     * @param String $s
     * @return String
     */
    function longestPalindrome($s) {
        $len = strlen($s);
        if ($len == 0) {
            return '';
        }
        $max = 1;
        $start = 0;
        for ($i = 0; $i < $len; $i++) {
            $left = $right = $i;
            while ($right + 1 < $len && $s[$right] == $s[$right+1]) {
                $right++;
            }
            while ($left >= 0 && $right < $len && $s[$right] == $s[$left]) {
                $left--;
                $right++;
            }
            $curLen = $right - $left - 1;
            if ($curLen > $max) {
                $start = $left + 1;
                $max = $curLen;
            }
        }
        return substr($s, $start, $max);
    }
}

File: go/5-longest-palindromic-substring.go
package main

func longestPalindrome(s string) string {
	len := len(s)
	if len == 0 {
		return s
	}
	max := 1
	start := 0
	for i := 0; i < len; i++ {
		var left, right int = i, i
		for right+1 < len && s[right+1] == s[right] {
			right++
		}
		for left >= 0 && right < len && s[right] == s[left] {
			left--
			right++
		}
		curLen := right - left - 1
		if curLen > max {
			max = curLen
			start = left + 1
		}
	}
	return s[start : start+max]
}
